### PR TITLE
tutorial purpose & architecture overview

### PR DESCRIPTION
**Pull request by flyingmachine**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,6 +72,146 @@ operations of generating and manipulating entity graphs.
 
 ## Tutorial
 
+In learning any new tool, I think it's useful to begin by learning the
+tool's purpose, then getting a high-level overview of the architecture
+and how it achieve's the tool's purpose. With those concepts in place,
```
Comment (codonnell-reify) [inline README.md]: achieves

File: README.md [hunk 2]
```diff
@@ -71,6 +72,146 @@ operations of generating and manipulating entity graphs.
 
 ## Tutorial
 
+In learning any new tool, I think it's useful to begin by learning the
```
Comment (codonnell-reify) [inline README.md]: I strongly agree with this approach.

File: README.md [hunk 3]
```diff
@@ -71,6 +72,146 @@ operations of generating and manipulating entity graphs.
...
+instances. It's rare that you'll interact with ent types directly.
+
+**Ent.** An ent is an instance of an ent type. They have a name (`:t0,
+:u0`, etc), and reference other ents. They're represented as nodes in
+the SG, with directed edges going from ents to the ents they
+reference; there's a directed edge from `:tl0` to `:u0` because `:tl0`
+references `:u0`.
+
+Ents can be associated with additional data - for example, a user ent
+can be associated with a map of user data generated by
```
Comment (codonnell-reify) [inline README.md]: Maybe something like "a map of user data, like name and email" so it's totally clear that you're not talking about relation data.

**Comment by flyingmachine** [comment]
hey man!!! I think this first section of the tutorial is ready :)

File: README.md [hunk 4]
```diff
@@ -71,6 +72,209 @@ operations of generating and manipulating entity graphs.
...
+
+When you're writing a test, it's important to be able to a) concisely
+express the values you're testing and b) easily figure out why a test
+is failing.
+
+At odds with these requirements is the fact that Specmonstah returns
+rich data structures under the philosophy that it's better to have
+information and not need it than need it and not have it. This can be
+overwhelming; picture two or three screens of output when you try to
+view a raw Specmonstah value in the REPL. But fear not: Specmonstah
```
[the remark this answers is not in the document]
Comment (flyingmachine) [reply README.md]: I wasn't trying to distinguish the two because I hadn't introduced the db concept yet

**Comment by flyingmachine** [comment]
I like your suggestions! We could update the very first section of the README, _Purpose_, by including additional examples of spec generation and db insertion. I didn't want the initial code snippet to include actual specs because I think they'd distract from the code we want to showcase, but we can elide those as you've suggested and put a link to the full code example.

There are definitely more ways that we can improve the README - for example I think it would be helpful to diagram the _generate graph / visit ents_ process somehow. If you think that this PR is a step forward I think we should go ahead and merge it; unlike with code it's easy to change and we're not committing ourselves to anything that has long-term consequences (like an API), plus it's not officially released so the stakes are even lower.

**Pull request merged**